rubix-prestosql/src/main/java/com/qubole/rubix/prestosql/CachingPrestoS3FileSystem.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2019. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.prestosql;
+
+import com.qubole.rubix.core.CachingFileSystem;
+import io.prestosql.plugin.hive.s3.PrestoS3FileSystem;
+import org.apache.hadoop.conf.Configuration;
+
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * Created by stagra on 14/1/16.
+ */
+public class CachingPrestoS3FileSystem extends CachingFileSystem<PrestoS3FileSystem>
+{
+  public CachingPrestoS3FileSystem()
+  {
+    super();
+  }
+
+  private static final String SCHEME = "s3n";
```

Comment:
I think the SCHEME should be **s3a**. Correct me if I'm wrong.

Reply:
old code again, it is from the changes to support rubix scheme, i think it doesnt matter if we use s3n or s3a here as old systems maps them to a FS.